.github/workflows/all-static-checks.yaml

```diff
@@ -61,3 +61,52 @@ jobs:
         run: |
           pip install pyyaml
           python tests/sweep_framework/sweeps_workflow_verification.py
+    name: Static analysis
+  clang-tidy:
+    runs-on: ubuntu-latest
+    container: ghcr.io/tenstorrent/tt-metal/tt-metalium/ubuntu-22.04-amd64:latest
+    permissions:
+      pull-requests: write
+      # OPTIONAL: auto-closing conversations requires the `contents` permission
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+        submodules: "recursive"
+    - name: Set safe directory for Git
+      run: git config --global --add safe.directory $GITHUB_WORKSPACE
+    - name: Fetch base branch
+      run: |
+        git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+        git fetch --no-tags upstream "${{ github.event.pull_request.base.ref }}"
+    - name: Install clang-tidy
+      run: |
+        sudo apt-get update
+        sudo DEBIAN_FRONTEND=noninteractive apt-get install -y clang-tidy-17 python3.11 python3.11-venv python3.11-dev
+        sudo ln -s $(which clang-tidy-17) /usr/local/bin/clang-tidy
+    - name: Prepare compile_commands.json
+      run: |
+        ARCH_NAME=grayskull cmake -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_DISABLE_PRECOMPILE_HEADERS=ON
+      continue-on-error: true
+    - name: Create results directory
+      run: |
+        mkdir clang-tidy-result
+    - name: Analyze
+      run: |
+        git diff -U0 "$(git merge-base HEAD "upstream/${{ github.event.pull_request.base.ref }}")" | clang-tidy-diff-17.py -p1 -path build -export-fixes clang-tidy-result/fixes.yml
+    - name: Run clang-tidy-pr-comments action
+      uses: platisd/clang-tidy-pr-comments@v1.5.0
```

Comment:
Especially because it's an external thing, we should tie this to a SHA (we should do this for all our workflows) https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions